### ISSUE TITLE
PHPCS 3.4.0: Deprecate the WordPress.CodeAnalysis.EmptyStatement sniff

### ISSUE THREAD

**Issue by jrfnl** [WordPress/WordPress-Coding-Standards#1523]
## Rationale

PR #980 introduced the `WordPress.CodeAnalysis.EmptyStatement` sniff.
The same sniff was later pulled to PHPCS as `Generic.CodeAnalysis.EmptyPHPStatement`.

At long last, the upstream sniff has now been merged and will be included in the PHPCS 3.4.0 release.

This means we can defer to the upstream sniff once the minimum supported PHPCS version goes up.

## References
* Upstream issue: https://github.com/squizlabs/PHP_CodeSniffer/issues/1119
* Upstream (merged) PR: https://github.com/squizlabs/PHP_CodeSniffer/pull/1814

## Action Checklist
Once PHPCS 3.4.0 has been released and WPCS is ready/willing to up the minimum required PHPCS version:
- [ ] Update the PHPCS version requirements in the `composer.json` file.
- [ ] Update references to the WPCS minimum PHPCS version requirements in the Readme and wherever else needed.
- [ ] Deprecate the `WordPress.CodeAnalysis.EmptyStatement` sniff.
     - [ ] Make the sniff extend the upstream `Generic.CodeAnalysis.EmptyPHPStatement` sniff and turn it into an empty class.
    - [ ] Mark the sniff as deprecated.
- [ ] Once this is done, open a new issue as a reminder to remove the WP native sniff in the next major version of WPCS and to replace the ruleset reference in the `WordPress-Core` ruleset.
